common/scpi_cmds.c

```diff
@@ -61,6 +63,25 @@ static uint32_t scpi_cmd_get_scp_cap_tx_payload[] = {
 	0,
 };
 
+/*
+ * Handler/payload data for SCPI_CMD_SET_SYS_PWR: Set system power state.
+ */
+uint32_t
+scpi_cmd_set_sys_power_handler(uint32_t *rx_payload, uint16_t rx_size __unused,
+                               uint32_t *tx_payload __unused,
+                               uint16_t *tx_size __unused)
+{
+	uint8_t system_state = rx_payload[0] & BITMASK(0, 8);
+
+	if (system_state != SYSTEM_POWER_STATE_SHUTDOWN &&
```

Comment:
Can you provide further explanation of what you are checking for here? It appears that `system_state` has to meet shutdown, reboot, and reset conditions simultaneously in order to reset.

Reply:
DeMorgan's law :). This says "report an error if it is none of these three".